core/lib/spree/core/permalinks.rb

```diff
@@ -52,15 +52,11 @@ def save_permalink(permalink_value=self.to_param)
           permalink_value ||= generate_permalink
 
           field = self.class.permalink_field
-            # Do other links exist with this permalink?
-            other = self.class.where("#{self.class.table_name}.#{field} LIKE ?", "#{permalink_value}%")
-            if other.any?
-              # Find the existing permalink with the highest number, and increment that number.
-              # (If none of the existing permalinks have a number, this will evaluate to 1.)
-              number = other.map { |o| o.send(field)[/-(\d+)$/, 1].to_i }.max + 1
-              permalink_value += "-#{number.to_s}"
-            end
-          write_attribute(field, permalink_value)
+          any_exist_with_number = lambda { self.class.where("#{self.class.table_name}.#{field} = ?", "#{permalink_value}") }
```

Comment:
`permalink_value` doesn't need to be string interpolated anymore. Unless I'm mistaken, we can actually get rid of all the string interpolation here with something like:

```Ruby
any_exist_with_number = lambda do
  self.class.where(self.class.table_name => { field => permalink_value })
end
```